pkg/operator/ceph/controller/predicate.go

```diff
@@ -675,3 +625,8 @@ func IsDoNotReconcile(labels map[string]string) bool {
 
 	return false
 }
+
+func ReloadManager() {
+	p, _ := os.FindProcess(os.Getpid())
+	_ = p.Signal(syscall.SIGHUP)
```

Comment:
By sending the signal, that means the operator is effectively restarted? So all the controllers, goroutines, health watchers, etc, will all be shut down and have a chance to start cleanly? What happens to the operator log? Does the operator log start fresh as well or does it continue? I hope it continues.

Reply:
No no the signal is not an interrupt, it's a reload so the operator process remains alive it's the controller runtime manager that it restarted. The operator pod never restarts.
But you are bringing up a good point for the health goroutine, let me verify, perhaps I just need to pass the context further down.